src/backend/TrafficCourts/Citizen.Service/Services/OAuth/OAuthUserService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using StackExchange.Redis;
+using System.Text.Json;
+using System.Text;
+using System.Buffers;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Diagnostics;
+using TrafficCourts.Citizen.Service.Models.OAuth;
+using System.Configuration;
+
+namespace TrafficCourts.Citizen.Service.Services
+{
+    /// <summary>
+    /// Implentation of the IOAuthUserService that is used for retrieving user information from OAuth's services.
+    /// </summary>
+    public class OAuthUserService : IOAuthUserService
+    {
+        private readonly Uri _userInfoEndpoint;
+        private readonly ILogger<OAuthUserService> _logger;
+
+        public OAuthUserService(Configuration.OAuthOptions options, ILogger<OAuthUserService> logger)
+        {
+            ArgumentNullException.ThrowIfNull(options);
+            _userInfoEndpoint = options.UserInfoEndpoint ?? throw new ArgumentException($"{nameof(options.UserInfoEndpoint)} is required");
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public UserInfo? GetUserInfo<UserInfo>(string token)
+        {
+            try
+            {
+                using (HttpClient client = new HttpClient())

```

Comment:
You cannot use HttpClient this way. We need to use `HttpClientFactory`. Couple of problems using HttpClient in this way. They are documented on this page [Use IHttpClientFactory to implement resilient HTTP requests](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests)
1.